### PR TITLE
fix: Reconnect to Arcadia when connection lost.

### LINKED ISSUE #9: REPL disconnects when starting game in Unity
REPL connection is broken when starting the game in Unity.

Steps: 
- Connect REPL to Arcadia
- Play game in Unity
- Stop game in Unity
- Eval code in REPL

Happened:
`java.net.SocketException: The transport's socket appears to have lost its connection to the nREPL server` 

Also broken while game is running

### PR DESCRIPTION
**Pull request by dnegma**
Fixes #9

When running the game in Unity, the nREPL connection is lost and in order to continue send code to Unity one has to restart the bridge and the remote real in the project. 

This is fixing that by trying to reconnect when connection is lost, up to 5 retries.

### CODE REVIEW

File: src/eponai/tools/cursive_arcadia_repl.clj [hunk 1]
```diff
@@ -52,6 +60,18 @@
               (assoc :id id))))))))
 
 
+(defn arcadia-nrepl-eval [handler msg]
```
Comment (petterik) [inline src/eponai/tools/cursive_arcadia_repl.clj]: Minor: How about changing the names of these arcadia functions?
* `arcadia-handle-message` -> `arcadia-eval`
* `arcadia-nrepl-eval` -> `try-arcadia-eval`

File: src/eponai/tools/cursive_arcadia_repl.clj [hunk 2]
```diff
@@ -52,6 +60,18 @@
...
+  (try
+    (arcadia-handle-message handler msg)
+    (catch SocketException e
+      (print "Arcadia connection lost, reconnecting...")
+      (connect!)
```
Comment (petterik) [inline src/eponai/tools/cursive_arcadia_repl.clj:68]: How about surrounding the retry logic around this `connect!` function and add a sleep?

File: src/eponai/tools/cursive_arcadia_repl.clj [hunk 3]
```diff
@@ -70,13 +90,13 @@
   [& args]
   (let [{:keys [arcadia-port port]} (:options (cli/parse-opts args cli-opts))]
     (print (format "Connecting to Arcadia on port %d... " arcadia-port))
-    (with-open [conn (nrepl/connect :port arcadia-port)]
+    (binding [*arcadia-port* arcadia-port]
+      (connect!)
```
Comment (petterik) [inline src/eponai/tools/cursive_arcadia_repl.clj:92]: I wonder if we should isolate nREPL connection inside an object that handles the reconnection logic itself.

Something with a protocol like:
```clj
(defprotocol INREPLConn
  (connect! [this])
  (send-message! [this message]))
```

We'd get rid of one var and we could get rid of the `(binding [*arcadia-port* arcadia-port]` if the object is constructed with the port.

Does this make sense?

**Review by petterik — COMMENTED** [review]
@dnegma This is really nice to have! Left some comments

File: src/eponai/tools/cursive_arcadia_repl.clj (hunk 2 above)
Comment (dnegma) [reply src/eponai/tools/cursive_arcadia_repl.clj:68]: What do you mean we should surround it with?

Comment (dnegma) [reply src/eponai/tools/cursive_arcadia_repl.clj:68]: Talked in person, and decided this is out of scope. This PR is only fixing the problem with the REPL disconnecting when the game runs, and we don't need the retry logic for that.

File: src/eponai/tools/cursive_arcadia_repl.clj [hunk 4]
```diff
@@ -52,6 +60,18 @@
...
 
 
+(defn arcadia-nrepl-eval [handler msg]
+  (try
+    (arcadia-handle-message handler msg)
+    (catch SocketException e
+      (print "Arcadia connection lost, reconnecting...")
+      (connect!)
+      (println "Done!")
+      (if (> 5 (:attempts msg 0))
```
Comment (petterik) [inline src/eponai/tools/cursive_arcadia_repl.clj]: Remove attempt garbage? :p

**Review by petterik — APPROVED** [review]
LGTM. Fix the remaining things whenever you have the chance.

**Pull request merged**